packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts

```diff
@@ -4,20 +4,41 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import {expect} from '@loopback/testlab';
+import axios, {AxiosProxyConfig, AxiosRequestConfig} from 'axios';
 import delay from 'delay';
 import http from 'http';
 import {AddressInfo} from 'net';
 import pEvent from 'p-event';
 import path from 'path';
-import makeRequest from 'request-promise-native';
 import rimrafCb from 'rimraf';
+import tunnel, {ProxyOptions as TunnelProxyOptions} from 'tunnel';
+import {URL} from 'url';
 import util from 'util';
 import {HttpCachingProxy, ProxyOptions} from '../../http-caching-proxy';
 
 const CACHE_DIR = path.join(__dirname, '.cache');
 
 const rimraf = util.promisify(rimrafCb);
 
+const axiosInstance = axios.create({
+  // Provide a custom function to control when Axios throws errors based on
+  // http status code. Please note that Axios creates a new error in such
+  // condition and the original low-level error is lost
+  validateStatus: () => true,
+});
+
+const makeRequest = async (config: AxiosRequestConfig) => {
+  const res = await axiosInstance(config);
+  // Throw an error with message from the original error
+  if (res.status >= 300) {
+    const err = new Error(`${res.status} - ${res.data}`);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (err as any).status = res.status;
+    throw err;
+  }
+  return res;
```

Comment:
Have you considered enhancing `res` object with `body` and `statusCode` properties, so that we don't have to modify all tests below?

```suggestion
  return Object.assign(res, {statusCode: res.statusCode, body: res.data});
```